WordPress/Classes/Services/AccountService+Swift.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+import WordPressShared
+import ShareExtensionCore
+
+extension AccountService {
+    func setupAppExtensions() {
+        let context = coreDataStack.mainContext
+        context.performAndWait {
+            guard let account = try? WPAccount.lookupDefaultWordPressComAccount(in: context) else {
+                return
+            }
+            self.setupAppExtensions(defaultAccount: account)
+        }
+    }
+
+    @objc func setupAppExtensions(defaultAccount: WPAccount) {
+        let shareExtensionService = ShareExtensionService()
+        let notificationSupportService = NotificationSupportService()
+
+        guard let defaultBlog = defaultAccount.defaultBlog, !defaultBlog.isDeleted else {
+            DispatchQueue.main.async {
+                shareExtensionService.removeShareExtensionConfiguration()
+                notificationSupportService.deleteServiceExtensionToken()
+            }
+            return
+        }
+
+        let defaultAccountObjectID = TaggedManagedObjectID(defaultAccount)
+        let siteID = defaultBlog.dotComID?.intValue
+        let siteName = defaultBlog.settings?.name
+
+        DispatchQueue.main.async {
+            do {
+                let defaultAccount = try self.coreDataStack.mainContext.existingObject(with: defaultAccountObjectID)
+
+                if let siteID, let siteName {
+                    shareExtensionService.storeDefaultSiteID(siteID, defaultSiteName: siteName)
```

Comment:
I kept the implementation as close to the original as I could. The only difference: 

- Added `TaggedManagedObjectID`
- Added `guard`
- Update according to the nullability. 

Previously, all these methods would just crash when they get an optional:

```swift
public func storeToken(_ oauth2Token: String) {
```

Now, I added `wpAssertFailure` to track these in production if it ever happens.